frontend/src/abs/frontend/pardef/PartialFunctionExpander.jadd

```diff
@@ -54,8 +54,12 @@ aspect PartialFunctionExpander {
 
     public void ModuleDecl.expandChildParFnApps(ASTNode<?> node) {
         for (ParFnApp fnApp : node.findChildren(ParFnApp.class, true)) {
-            if (fnApp.closestParent(PartialFunctionDecl.class) != null) {
-                // ignore ParFnApp within pardef that has not been expanded yet
+            if (fnApp.closestParent(PartialFunctionDecl.class) != null
+                    || fnApp.closestParent(TraitDecl.class) != null) {
```

Comment:
Would it be possible to express these in terms of `canBeRewritten()` / `amInCoreAbs()`?  See file `src/abs/backend/coreabs/GenerateCoreAbs.jadd` for how this is handled for `ForEachStmt`s

Reply:
I.e., with `amInCoreAbs()`, partial functions should be left alone when in the body of a delta as well, and in future constructs that might be introduced.

Reply:
Adding `amInCoreAbs()` as an additional condition is probably a good idea, but it wouldn't solve the problem on its own, because it does not return false for children of a `TraitDecl` (despite the code comments saying otherwise) and I'm somewhat hesitant to change that.

Additionally, this code specifically looks for `ParFnApp` children of `ModuleDecl`, so AFAIK checking for a delta parent is not necessary.

Reply:
Good point; Traits are in fact in core ABS, so the current behavior of `amInCoreAbs()` is correct.  Not expanding outside of a "proper" module is the way to go, regardless of how you express it.  Just trying to reduce the number of idioms we use to express this, but it's a bit up to taste.

Reply:
For the record: I absolutely agree with you on only defining this sort of thing once (in this case in `amInCoreAbs()`) and using it everywhere as it definitely increases maintainability.